avocado/core/nrunner.py

```diff
@@ -122,6 +122,7 @@ def __init__(self, kind, uri, *args, config=None, **kwargs):
         self.tags = kwargs.pop('tags', None)
         self.requirements = kwargs.pop('requirements', None)
         self.variant = kwargs.pop('variant', None)
+        self.output_dir = kwargs.pop('output_dir', None)
```

Comment:
I understand your concern abut this, but I don't think that usage of config is a good solution here. The configuration is set by user, and it is something which shouldn't by changed by avocado itself during runtime. We have the same problem with `tags`, `requirments` and `variants` so maybe we should come up with different solution for sending those data to runners.

Reply:
My problem with this approach is that we teach developers and users saying that `args` and `kwargs` are handled different by runners and usually. In the case of exectest, `kwargs` are environment variables passed to the tests. However here, we have to keep a "reserved words" list where those "environment variables"  can't be used because we are using internally, and we are just increasing this list.
  
About the configuration aspect: users can configure things based on `avocado.conf` or using the config dict (when using the Job API). The entire idea behind the config dict was that we could concentrate settings in a single place, to avoid confusion and here we are still using a different method to configure things. IMO this could be solved using the same dict, by using different namespaces: `runnable.*` or `runners.*` for instance. And we could even state that some namespaces would be read only, because they are used internally.

Also, due our architecture, third-party plugins wouldn't need to change core method's signature to support new features, they could rely on some config's namespace.

Btw, this is not a blocker, but I think we should think about it to avoid future issues.